rmw_fastrtps_shared_cpp/src/custom_publisher_info.cpp

```diff
@@ -138,6 +178,32 @@ void PubListener::set_on_new_event_callback(
           incompatible_qos_status_.total_count_change = 0;
         }
         break;
+      case RMW_EVENT_PUBLICATION_MATCHED:
+        {
+          publisher_info_->data_writer_->get_publication_matched_status(
+            matched_status_);
```

Comment:
it shouldn't be triggered the event callback for previous events.
It makes sense for messages, but IMO not for new matches.

Reply:
So you mean we should directly use matched_status_.

Reply:
I mean that the callback shouldn't be called when it's added.
i.e. do nothing in this listener.

Though we're already doing something for all the other events, which I'm not sure if it's a good idea or not, but we're already doing.
So in that case, yes using `matched_status_` would also be okay.